src/commands/InitCommand.ts

```diff
@@ -695,7 +695,10 @@ Steps to run this project:
 
         if (!packageJson.dependencies) packageJson.dependencies = {}
         Object.assign(packageJson.dependencies, {
-            typeorm: require("../package.json").version,
+            typeorm:
+                require("../package.json").version !== "0.0.0"
+                    ? require("../package.json").version // install version from package.json if present
+                    : require("../package.json").installFrom, // else use custom source
```

Comment:
The tests fail because the implemented "fix" did not comply to the [package.json specs](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#version:~:text=Version%20must%20be%20parseable%20by%20node%2Dsemver) which require the version to be parseable by semver (format x.x.x).
For some reason node v16 just did not care but v14 does.
The alternative would be to change the cli source code a bit to make detectable wether running in test mode or not.
I see no other solution for testing the desired behavoir with 100% code coverage.